pydra/engine/helpers.py

```diff
@@ -649,12 +649,9 @@ def argstr_formatting(argstr, inputs, value_updates=None):
     if value_updates:
         inputs_dict.update(value_updates)
     # getting all fields that should be formatted, i.e. {field_name}, ...
-    inp_fields = re.findall(r"{\w+}", argstr)
-    inp_fields_float = re.findall(r"{\w+:[0-9.]+f}", argstr)
-    inp_fields += [re.sub(":[0-9.]+f", "", el) for el in inp_fields_float]
+    inp_fields = re.findall(r"{(\w+)(?::[0-9.]+f|\[[\w]+\])?}", argstr)
```

Comment:
IMO we should support the whole range of keyword-arg format strings. Reading [PEP 3101](https://peps.python.org/pep-3101/#standard-format-specifiers), we can get a pretty good approximation:

```python
def parse_format_string(fmtstr):
    identifier = r"[a-zA-Z_]\w*"
    attribute = fr"\.{identifier}"
    item = fr"\[\w+\]"
    # Example: var.attr[key][0].attr2 (capture "var")
    field_with_lookups = f"({identifier})(?:{attribute}|{item})*"  # Capture only the keyword
    conversion  = "(?:!r|!s)"
    nobrace = "[^{}]*"
    # Example: 0{pads[hex]}x (capture "pads")
    fmtspec = f"{nobrace}(?:{{({identifier}){nobrace}}}{nobrace})?"  # Capture keywords in spec
    full_field = f"{{{field_with_lookups}{conversion}?(?::{fmtspec})?}}"

    all_keywords = re.findall(full_field, fmtstr)
    return set().union(*all_keywords) - {""}
```

This can be fully unit tested with a bunch of specifiers like:

```
{a}  # a
{abc}  # abc
{a:{b}} # a, b
{a:{b[2]}}  # a, b
{a.xyz[somekey].abc:{b[a][b].d[0]}}  # a, b
{a:05{b[a 2][b].e}}  # a, b
{a1_field} {b2_field:02f} -test {c3_field[c]} -me {d4_field[0]}  # a1_field, b2_field, c3_field, d4_field
```

Then:

```suggestion
    inp_fields = parse_format_string(argstr)
```